lib/charms/mongodb_libs/v0/mongodb_provider.py

```diff
@@ -56,7 +72,16 @@ def __init__(self, charm):
         self.framework.observe(self.charm.on[REL_NAME].relation_joined, self._on_relation_event)
         self.framework.observe(self.charm.on[REL_NAME].relation_changed, self._on_relation_event)
         self.framework.observe(self.charm.on[REL_NAME].relation_broken, self._on_relation_event)
+        self.framework.observe(
+            self.charm.on[LEGACY_REL_NAME].relation_created, self._on_legacy_relation_created
+        )
+        self.framework.observe(
+            self.charm.on[LEGACY_REL_NAME].relation_joined, self._on_legacy_relation_joined
+        )
```

Comment:
suggestion: move all legacy code to a separate lib-like file (so it is easy to deprecate and rip out of the code base when its time comes)

Reply:
Yes I'll keep that in mind for the organisational follow up PR! :)